### PR TITLE
RW-12082 add more tags for some existing metrics

### PR DESCRIPTION
**Pull request by Qi77Qi**
https://precisionmedicineinitiative.atlassian.net/browse/RW-12082

I'm looking to build a dashboard for metrics that AoU team care about, and notice for some metrics, we can add a bit more info via adding more tags. Also added one for tracking failed async tasks

__Jira ticket__: https://broadworkbench.atlassian.net/browse/[ticket_number]

### What

-

### Why

-

### What to test

- 

### Who tested and where

- [ ] This change is covered by automated tests 
  - _NB: Rerun automation tests on this PR by commenting `jenkins retest` or `jenkins multi-test`._
- [ ] I validated this change 
- [ ] Primary reviewer validated this change 
- [ ] I validated this change in the dev environment

### CODE REVIEW

File: core/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/AsyncTaskProcessor.scala [hunk 1]
```diff
@@ -29,15 +29,15 @@ final class AsyncTaskProcessor[F[_]](config: AsyncTaskProcessor.Config, asyncTas
...
+      tags = Map("taskName" -> task.taskName) ++ task.additionalMetricsTags
       _ <- recordLatency("asyncTaskLatency", latency, tags)
       _ <- logger.info(Map("traceId" -> task.traceId.asString))(
         s"Executing task ${task.taskName} with latency of ${latency.toSeconds} seconds"
       )
       _ <- task.op.handleErrorWith { case err =>
         task.errorHandler.traverse(cb => cb(err)) >> logger.error(Map("traceId" -> task.traceId.asString), err)(
           s"Error when executing async task"
-        )
+        ) >> metrics.incrementCounter("asyncTaskError", 1, tags)
```
Comment (Qi77Qi) [inline core/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/AsyncTaskProcessor.scala]: this is new metrics for tracking failed async tasks

**Review by yonghaoy — APPROVED** [review]

**Review by LizBaldo — APPROVED** [review]

**Comment by Qi77Qi** [comment]
@LizBaldo  I pushed a bit more changes...PTAL...the main change is I notice we can't add tags to the same metrics in a way that events may have different set of tags. Hence I added a `TaskMetricsTag` so that we'll always consistently add these tags, and for ones that don't apply to certain events, we'll default to `unknown`.

Another change I made is we aren't naming events consistently in [AzurePubsubHandler.scala](https://github.com/DataBiosphere/leonardo/pull/4341/files#diff-e140eb1deb8547995938a18b2bea7c0a97756d55ca5f62d3fa7592a99faabaa4), so updated `createAzureRuntime` metrics to `createRuntimeV2` to be consistent with other ones. Also noticed a copy-paste error in the metrics name for stopRuntimeV2 event

**Comment by Qi77Qi** [comment]
```
[info] - should have the workspace-related environment variables set in jupyter image *** FAILED *** (28 seconds, 339 milliseconds)
[info]   java.lang.RuntimeException: Nonzero exit value: 255
```

jenkins retest

**Comment by Qi77Qi** [comment]
@LizBaldo  it's been a while since u approved last time...can u take another look

**Comment by LizBaldo** [comment]
LGTM, :shipit:

**Pull request merged**